ResearchKit/Common/ORKImageCaptureView.m

```diff
@@ -197,31 +221,15 @@ - (void)setSession:(AVCaptureSession * __nullable)session {
     [self orientationDidChange];
 }
 
-- (UIBarButtonItem *)continueButtonItem {
-    return _continueSkipContainer.continueButtonItem;
-}
-
-- (void)setContinueButtonItem:(UIBarButtonItem *)continueButtonItem {
-    // Intercept the continue button press.  This can be called multiple
-    // times with the same UIBarButtonItem, or with different UIBarButtonItems,
-    // so capture it whenever the button does not point to our selector
-    if (continueButtonItem.action != @selector(capturePressed)) {
-        self.continueAction = continueButtonItem.action;
-        self.continueTarget = continueButtonItem.target;
-        self.continueTitle = continueButtonItem.title;
-        continueButtonItem.action = @selector(capturePressed);
-        continueButtonItem.target = self;
-    }
+- (void)setSkipButtonItem:(UIBarButtonItem *)skipButtonItem {
+    if (_showSkipButtonItem) {
+        _skipButtonItem = skipButtonItem;
     
-    // If we do not have an error to show, and we haven't already gotten a captured
-    // image, then change the title of the button to be appropriate for the capture action
-    if (!self.capturedImage && !self.error) {
-        continueButtonItem.title = ORKLocalizedString(@"CAPTURE_BUTTON_CAPTURE_IMAGE", nil);
-    } else if(self.capturedImage) {
-        _continueSkipContainer.skipButtonItem = _skipButtonItem;
+        // If the recapture button is not currently being used as the skip button, then use this new button
+        if (_continueSkipContainer.skipButtonItem != _recaptureButtonItem) {
+            _continueSkipContainer.skipButtonItem = skipButtonItem;
+        }
     }
-    
-    _continueSkipContainer.continueButtonItem = continueButtonItem;
 }
 
```

Comment:
The ``ORKStepViewController.setupButtons`` function sets a non-nil skipButtonItem on ``ORKImageCapturetepViewController`` no matter what ``optional`` is set to in the ``ORKImageCaptureStep``.  Therefore, this check is to ensure that we only show the VC provided skip button item in the ``ORKNavigationContainerView`` inside the ``ORKImageCaptureView`` if optional is YES.

I can add the continue button item setter if you like.